### ISSUE TITLE
Images not showing (403 error) on private simple sites with a custom primary domain. 

### ISSUE THREAD

**Issue by Greatdane** [Automattic/wp-calypso#90972]
### Quick summary

When a Simple Site is set to private, media images do not show up in Calypso interfaces such as the Media library (or blocks dependent on it like the [Slideshow Block](https://wordpress.com/support/wordpress-editor/blocks/slideshow-block/)) 

Browser console shows;

```
GET https://i0.wp.com/domain.com/wp-content/uploads/2024/04/20240224 181813.jpg?ssl=1&resize=438%2C438 403
(Forbidden)
```

### Steps to reproduce

1. Have a simple site with a custom domain set to private
2. Go to the media library (Calypso) or add a slideshow block and try and choose images
3. The images will not load with a 403 error

### What you expected to happen

Images should show

### What actually happened

Images show a 403 error

### Impact

All

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple

### Logs or notes

- Can switch to classic view in media library, but cannot in slideshow block. 
- If you switch back to `.wordpress.com` as primary domain, the images load
- If you make the site public, images load.

**Comment by devicenull** [comment]
The fix in #87642 should resolve this as well

**Comment by liviopv** [comment]
📌 REPRODUCTION RESULTS
- Tested on Simple – Replicated
- 
📌 FINDINGS/SCREENSHOTS/VIDEO
![CleanShot 2024-05-23 at 11 32 22@2x](https://github.com/Automattic/wp-calypso/assets/14153300/8fdf89a7-7b38-474a-ab27-74d58d063c07)

📌 ACTIONS
- Triaged

@devicenull do we need to keep this issue or is it ok to close it as dupe of https://github.com/Automattic/wp-calypso/issues/87642?
